homeassistant/components/somfy/climate.py

```diff
@@ -0,0 +1,214 @@
+        domain_data = hass.data[DOMAIN]
+        coordinator = domain_data[COORDINATOR]
+        api = domain_data[API]
+
+        return [
+            SomfyClimate(coordinator, device_id, api)
+            for device_id, device in coordinator.data.items()
+            if SUPPORTED_CATEGORIES & set(device.categories)
+        ]
+
+    async_add_entities(await hass.async_add_executor_job(get_thermostats))
+
+
+class SomfyClimate(SomfyEntity, ClimateEntity):
+    """Representation of a Somfy thermostat device."""
+
+    def __init__(self, coordinator, device_id, api):
+        """Initialize the Somfy device."""
+        super().__init__(coordinator, device_id, api)
+        self.climate = None
+        self._create_device()
+
+    def _create_device(self):
+        """Update the device with the latest data."""
+        self.climate = Thermostat(self.device, self.api)
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+        return supported_features
+
+    @property
+    def device_state_attributes(self) -> Dict[str, Any]:
+        """Return the state attributes of the device."""
+        return {ATTR_BATTERY_LEVEL: self.climate.get_battery()}
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.climate.get_ambient_temperature()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self.climate.get_target_temperature()
+
+    def set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+
+        self.climate.set_target(TargetMode.MANUAL, temperature, DurationType.NEXT_MODE)
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return 26.0
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return 15.0
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self.climate.get_humidity()
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if self.climate.get_regulation_state() == RegulationState.TIMETABLE:
+            return HVAC_MODE_AUTO
+        return HVAC_MODES_MAPPING.get(self.climate.get_hvac_state())
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        hvac_state = HVAC_MODES_MAPPING.get(self.climate.get_hvac_state())
+        return [HVAC_MODE_AUTO, hvac_state]
```

Comment:
This returns two possible modes: AUTO, and the current state.  Should this instead return all supported modes (values) from HVAC_MODES_MAPPING independent of the current device state similar to preset_modes below?

Reply:
I agree it looks confusing. And I got it once I was able to manipulate the Somfy application. 
The Somfy thermostat has the AUTO mode (called timetable, a scheduler), and COOL or HEAT mode. These two last modes are exclusive. You have to update manually the application to switch between cool and heat. And, your climate device has to support of course the cool mode. 
So to avoid exposing a mode that's not available, I code what you just commented. I don't want to show cool mode if the device does not support it.

Reply:
I will add it.

Reply:
We shouldn't include `None` in the hvac modes.

Reply:
hvac_state cannot be None. Do you want I raise an error just in case it happens?

Reply:
Then we shouldn't use `dict.get`.